src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -347,10 +296,118 @@ HTTPProxy::_handleGetSchema(shd::GetSchemaRequest&& request) {
     });
 }
 
+void HTTPProxy::setQueryRecord(const sh::String& collectionName, shd::SKVRecord::Storage&& key, dto::SKVRecord& k2record) {
+    if (key.fieldData.size() == 0) return;
+    auto shdSchema = getSchemaFromCache(collectionName, k2record.schema);
+    shd::SKVRecord shdrecord(collectionName, shdSchema, std::move(key), true);
+    _shdRecToK2(shdrecord, k2record);
+}
+
+namespace dtoexp = dto::expression;
```

Comment:
a 'k2exp' would be better here since we want to distinguish k2 vs sh expressions